config/v3_3_experimental/config.go

```diff
@@ -55,3 +57,22 @@ func Parse(rawConfig []byte) (types.Config, report.Report, error) {
 
 	return config, rpt, nil
 }
+
+// ParseCompatibleVersion parses the raw config of version 3.3.0-experimental or
+// lesser into a 3.3-exp types.Config struct and generates a report of any errors,
+// warnings, info, and deprecations it encountered
+func ParseCompatibleVersion(raw []byte) (types.Config, report.Report, error) {
+	version, rpt, err := util.GetConfigVersion(raw)
+	if err != nil {
+		return types.Config{}, rpt, err
+	}
+
+	if version == types.MaxVersion {
+		return Parse(raw)
+	}
+	cfg, r, err := v3_2.ParseCompatibleVersion(raw)
```

Comment:
For clarity, I'd make the first variable something like `prevCfg`, since it's not a config for the current version.